### PR TITLE
DAS-2214: Add the pre-commit file and update readme.

### PR DESCRIPTION
**Pull request by flamingbear**
## Description

Adds pre-commit functionality to the harmony-regression-tests repository
 
When pre-commit.ci is enabled: this will automatically run our pre-commit hooks for each PR. 
Rather than fixing errors, it will simply fail the PR checks.

If a developer installs the pre-commit hooks themselves, they will catch errors
before they are even committed, again no fixes are automatically applied, but
the files are updated to be correct and the user can add the changes to their commit.

Additionally, I have bumped every package version by a patch version in order
to release the changes to the code. There should be no difference in how the
tests function.  I have verified this but also included instructions in the
test for you do to the same.

## Jira Issue ID

DAS-2214 ostensibly but mostly because I have to do significant ipython notebook writing.

## Local Test Steps

Check that the commit in the `.git-blame-ignore-revs` file contains all of the reformatting changes. 

Pull this branch.
Follow the new instructions in the README under `pre-commit hooks`. 

To test this, make some minor change to a .py file or an .ipynb file, use git
to add the changes to a commit. You should see the checks all run when you are
trying to use git commit.

To test the all of the code changes / bumped versions. 

from the `test` directory make and run all of the tests.

For your UAT environment set all of these variables:
export HARMONY_HOST_URL=https://harmony.uat.earthdata.nasa.gov
export AWS_ACCESS_KEY_ID=
export AWS_SECRET_ACCESS_KEY=
export EDL_USER=
export EDL_PASSWORD=

Then make all images with: 

``` shell
> make images
```

That will take a while  and then run them:

``` shell
./run_notebooks.sh
```

Ensure all tests pass 

## PR Acceptance Checklist
* [N/A] Acceptance criteria met
* [N/A] Tests added/updated (if needed) and passing
* [X] Documentation updated (if needed)

### CODE REVIEW

**Comment by flamingbear** [comment]
Mostly for awareness: @chris-durbin @ygliuvt @indiejames 

This just adds auto-formatting via pre-commit hooks to our jupyter notebooks and python modules. But since you also work in here sometimes. I'm pinging you.

File: test/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,4 @@
-# Uses focal image because newer jammy image does not work with old docker versions
-# see related information: https://github.com/adoptium/containers/issues/215#issuecomment-1142046045
-FROM mambaorg/micromamba:1.4.2-focal
+FROM mambaorg/micromamba:1.5.8-jammy
```
Comment (flamingbear) [inline test/Dockerfile:1]: This updates the base image, but the old version was only because we couldn't update docker on the bamboo agents.

Comment (flamingbear) [reply test/Dockerfile:1]: It does seem to fix some vulnerabilities as well. 
Screenshot 2024-08-05 at 1 46 31 PM
Screenshot 2024-08-05 at 1 45 46 PM

File: README.md [hunk 2]
```diff
@@ -248,3 +251,36 @@ if the new image is named `ghcr.io/nasa/regression-tests-foo`, then we would add
...
+* Ensure JSON files have valid formats.
+* [ruff](https://github.com/astral-sh/ruff) Python linting checks.
+* [black](https://black.readthedocs.io/en/stable/index.html) Python code
+  formatting checks.
+
+To enable these checks:
+
+```bash
+# Install pre-commit Python package as part of test requirements:
+pip install -r dev-requirements.txt
```
Comment (owenlittlejohns) [inline README.md:271]: Should this file be included in the repository? I don't think I see it.

Comment (flamingbear) [reply README.md]: That's probably copy-paste error from earthdata-varinfo. 😳 There isn't a single python for this project so I just updated to install the pre-commit directly.

**Review by owenlittlejohns — APPROVED** [review]
Thanks for fixing all of this up and adding `pre-commit`. I ran all the regression tests locally, and they still pass. I also installed `pre-commit` via the new instructions, and it works as it should.

Nice!

**Comment by flamingbear** [comment]
Close and reopen to see if I can pickup the pre-commit.ci

**Comment by flamingbear** [comment]
reopening

**Pull request merged**